RecoEgamma/EgammaElectronProducers/plugins/LowPtGsfElectronSeedProducer.cc

```diff
@@ -0,0 +1,318 @@
+#include "TrackingTools/Records/interface/TransientRecHitRecord.h"
+#include "TrackingTools/TrackFitters/interface/TrajectoryFitter.h"
+#include "TrackingTools/TrajectoryParametrization/interface/GlobalTrajectoryParameters.h"
+#include "TrackingTools/TrajectoryState/interface/TrajectoryStateOnSurface.h"
+#include "TMath.h"
+
+LowPtGsfElectronSeedProducer::LowPtGsfElectronSeedProducer( const edm::ParameterSet& conf, 
+							    const lowpt::HeavyObjectCache* ) :
+  tracks_()
+{
+  tracks_ = consumes<reco::TrackCollection>( conf.getParameter<edm::InputTag>("tracks") );
+  clusters_ = consumes<reco::PFClusterCollection>( conf.getParameter<edm::InputTag>("clusters") );
+  fitter_ = conf.getParameter<std::string>("Fitter");
+  smoother_ = conf.getParameter<std::string>("Smoother");
+  builder_ = conf.getParameter<std::string>("TTRHBuilder");
+  pass_through_ = conf.getParameter<bool>("PassThrough");
+  produces<reco::ElectronSeedCollection>();
+  produces<reco::PreIdCollection>();
+}
+
+LowPtGsfElectronSeedProducer::~LowPtGsfElectronSeedProducer() {}
+
+void LowPtGsfElectronSeedProducer::beginLuminosityBlock( edm::LuminosityBlock const&, 
+							 edm::EventSetup const& setup ) 
+{
+  setup.get<IdealMagneticFieldRecord>().get(field_);
+}
+
+void LowPtGsfElectronSeedProducer::produce( edm::Event& event, 
+					    const edm::EventSetup& setup ) 
+{
+
+  auto seeds = std::make_unique<reco::ElectronSeedCollection>();
+  auto preids = std::make_unique<reco::PreIdCollection>();
+  
+  edm::Handle<reco::TrackCollection> tracks;
+  event.getByToken(tracks_, tracks);
+  
+  edm::Handle<PFClusters> clusters;
+  event.getByToken(clusters_,clusters);
+
+  for ( unsigned int itrk = 0; itrk < tracks.product()->size(); itrk++ ) {
+
+    reco::TrackRef track_ref(tracks,itrk);
+    if ( !(track_ref->quality(reco::TrackBase::qualityByName("highPurity"))) ) { continue; }
+
+    // Create ElectronSeed 
+    reco::ElectronSeed seed( *(track_ref->seedRef()) );
+    seed.setCtfTrack(track_ref);
+    
+    // Create PreId
+    reco::PreId preid;
+
+    // Add track ref to PreId
+    preid.setTrack(track_ref);
+
+    // Add variables related to track-ECAL matching to PreId
+    propagate_to_ecal(preid,track_ref,clusters);
+
+    // Add variables related to GSF tracks to PreId
+    light_gsf_tracking(preid,track_ref,seed,setup); 
+
+    // Decision based on BDT 
+    bool bdt = pass_through_ || bdt_decision(preid);
+
+    // Store PreId
+    preids->push_back(preid);
+
+    // 
+    if ( !bdt ) { continue; }
+    
+    seeds->push_back(seed);
+
+  }
+
+  event.put(std::move(seeds));
+  event.put(std::move(preids));
+
+}
+
+bool LowPtGsfElectronSeedProducer::propagate_to_ecal( reco::PreId& preid,
+						      const reco::TrackRef& track_ref,
+						      const edm::Handle<PFClusters>& clusters )
+{
+
+  // Propagate 'electron' to ECAL surface
+  float energy = sqrt( pow(0.000511,2.) + track_ref->outerMomentum().Mag2() );
```

Comment:
is this really worth this? The min p of a track is will be ~0.7 GeV assuming all in the transverse plane. 0.511 MeV added in quadrature is nothing as we are just wasting cpu.

Reply:
I can remove this but I still need to sqrt the outerMomentum().Mag2() to set the energy...?